src/mediapipe_internal/mediapipefactory.cpp

```diff
@@ -45,17 +45,18 @@ Status MediapipeFactory::createDefinition(const std::string& pipelineName,
     }
     std::shared_ptr<MediapipeGraphDefinition> graphDefinition = std::make_shared<MediapipeGraphDefinition>(pipelineName, config, manager.getMetricRegistry(), &manager.getMetricConfig());
     auto stat = graphDefinition->validate(manager);
```

Comment:
Validate does not check if pipeline or model already exitsts with such name. Validate does not use `manager` parameter at all. PipelineDefinition::validate checks such things.

Reply:
Existence of models is checked in ModelManager. We will need model manager here to check for availability of models.